### ISSUE TITLE
BUG: TUI animations do not work (i.e. progress bar & spinners from Rich library)

### ISSUE THREAD

**Issue by Emasoft** [holzschu/a-shell#537]
TUI animations do not work (i.e. progress bar & spinners from [Rich](https://github.com/Textualize/rich) library). 
They stand still and then update only after CTRL-C.

![IMG_0628](https://user-images.githubusercontent.com/713559/212207597-5b46ef21-7341-4d8e-8ab4-3e6cd0cf0405.gif)

**Comment by Emasoft** [comment]
This means that the sleep command stops all the threads in a-shell before the display is updated? Maybe the display updates is too slow? I've encountered a similar problem when using the console print and exiting before flushing the buffer.
By default, the python function print() prints to sys.stdout. Maybe the stdout and stderr streams buffer in a-shell is too slow or too big?

**Comment by holzschu** [comment]
> This means that the sleep command stops all the threads in a-shell before the display is updated?

Yes, I think that's exactly what is happening. sleep works on the entire process, and because it's iOS everything shares the same process. I haven't found a way to sleep a thread while the other threads keep going.

**Comment by Emasoft** [comment]
A temporary solution would be to make a-Shell force the Python interpreter to always flush the buffer before executing the next command (in this case sleep), so the display will be updated before everything freezes.
This can be done adding this environmental setting:
```
apply this to every python process in your environment with PYTHONUNBUFFERED=TRUE
```

**Comment by holzschu** [comment]
I tried (you can try it too) and it doesn't change anything. I don't think the issue is with the display.

**Comment by Iamrodos** [comment]
I have the same problem with rich and spinners. Normal rich text display (colors, tables etc) works fine. 

For example, this normally works to display a spinner whilst data is being pulled from a remote source.

```
with Status("Loading data ...", spinner_style="bold default"):
        my_data = load_data()
```

There are no occurrences of sleep within the code. In a-shell I just get a blank line instead of the spinner followed by the text.

I did think it was a threading issue as I believe Rich uses threads to update spinners and progress bars whilst your code executes. However, I ran a simple test of threading and it appears to work ok.

**Comment by Emasoft** [comment]
@Iamrodos :
> In a-shell I just get a blank line instead of the spinner followed by the text.

Same here. This is a pretty serious issue, since it makes Rich and Textual unusable in most scenarios.
